src/Test/Utilities/Shared/Pdb/PdbValidation.cs

```diff
@@ -143,14 +143,20 @@ public static class PdbValidation
 
         private static void RemoveEmptyScopes(XElement pdb)
         {
-            var emptyScopes = from e in pdb.DescendantsAndSelf()
-                              where e.Name == "scope" && !e.HasElements
-                              select e;
+            XElement[] emptyScopes;
 
-            foreach (var e in emptyScopes.ToArray())
+            do
             {
-                e.Remove();
+                emptyScopes = (from e in pdb.DescendantsAndSelf()
+                               where e.Name == "scope" && !e.HasElements
+                               select e).ToArray();
+
+                foreach (var e in emptyScopes)
+                {
+                    e.Remove();
+                }
             }
+            while (emptyScopes.Any());
```

Comment:
Under what conditions do we repeat the ```do...while```?  (naively, it seems like we'll only ever execute this once)